erdjava/src/main/java/elrond/Address.java

```diff
@@ -0,0 +1,68 @@
+package elrond;
+
+import org.bouncycastle.util.encoders.Hex;
+
+import elrond.Bech32.Bech32Data;
+import elrond.Exceptions.AddressException;
+
+public class Address {
+    static final String HRP = "erd";
+    static final int PUBKEY_LENGTH = 32;
+    static final int PUBKEY_STRING_LENGTH = PUBKEY_LENGTH * 2; // hex-encoded
+    static final int BECH32_LENGTH = 62;
+
+    private final String valueHex;
+
+    private Address(String valueHex) {
+        this.valueHex = valueHex;
+    }
+
+    public static Address createEmptyAddress() {
+        return new Address("");
+    }
+
+    public static Address fromBech32(String value) throws Exceptions.AddressException {
+        Bech32Data bech32Data = Bech32.decode(value);
+        if (!bech32Data.hrp.equals(HRP)) {
+            throw new Exceptions.BadAddressHrpException();
+        }
+
+        byte[] decodedBytes = Bech32.convertBits(bech32Data.data, 5, 8, false);
+        String hex = new String(Hex.encode(decodedBytes));
+        return new Address(hex);
+    }
+
+    public static Address fromHex(String value) throws Exceptions.AddressException {
+        byte[] decoded = Hex.decode(value);
```

Comment:
Is this because Hex.decode("invalid hex") will return empty string? In that case, can you pull this into another function? We could also check for the length of the address, right?

Reply:
Actually, it seems that:

```
byte[] foo = Hex.decode("foobar");
byte[] foo = Hex.decode("foo");
```

will throw a `org.bouncycastle.util.encoders.DecoderException` exception. Thanks, will fix!